### ISSUE TITLE
NodeJS breakpoint won't stop with capabilities set

### ISSUE THREAD

**Issue by clarkttfu** [microsoft/vscode-js-debug#852]
## Config

OS (VmWare Fusion virtual machine)
Debian 4.19.98-1 (2020-01-26) x86_64 GNU/Linux
Nodejs 12.14.1 / 12.19.0 / 14.15.0

## Problem

1. setup capability: `sudo setcap cap_net_admin+iep $(which node)`
2. launch debug session and break point stops
3. quit vscode and kill the remote vscode-server process then start vscode remote dev session
4. launch debug session and break point **do not** stop

[product_json.txt](https://github.com/microsoft/vscode-remote-release/files/5451534/product_json.txt)

FYI:
https://man7.org/linux/man-pages/man7/capabilities.7.html

**Comment by connor4312** [comment]
I'm not familiar at all with Linux capabilities. Just trying to reproduce this using the instructions you gave, I'm no longer able to execute node. Can you provide more detailed steps to help me repro?

```
$ sudo setcap cap_net_admin+iep $(which node)
$ ls -l $(which node)
-rwxrwxrwx 1 root staff 73617512 Jul 29  2020 /usr/local/bin/node
$ node --version
bash: /usr/local/bin/node: Operation not permitted
$ sudo node --version
sudo: unable to execute /usr/local/bin/node: Operation not permitted
```

**Comment by clarkttfu** [comment]
Hi, thanks for the update!

I think the problem is that you used the global node executable, which usually is installed by root...
I use [nvm](https://github.com/nvm-sh/nvm) to install and use node executable inside ~/.nvm.

With nvm, you can install and manage different versions of node for yourself, without affecting other users on the same host. But to set capability, `sudo` is required to execute `setcap`.

More info about capability:

For Debian you may need package `libpam-cap` and add a line (before the line `none  *`) like below into /etc/security/capability.conf 
```
cap_sys_admin,cap_net_admin            your_user_name_here
```

**Comment by connor4312** [comment]
Thanks, I was originally using a container to repro on windows but it looks like setcap is broken (at least this capability) there.

This looks like it's actually a bug in Node.js, I've opened https://github.com/nodejs/node/issues/37588

As a workaround you can launch your programs directly in the terminal using `--inspect` and then [attach](https://code.visualstudio.com/docs/nodejs/nodejs-debugging#_attaching-to-nodejs) to them.

**Comment by clarkttfu** [comment]
> This looks like it's actually a bug in Node.js

Thanks for the update, but would you please share some detail here?

**Comment by connor4312** [comment]
What detail do you have questions about?

**Comment by clarkttfu** [comment]
As I can recall it was good in the beginning of 2020 and it was some old version of node 12. Does it relate to the node version used by vscode remote server?

**Comment by connor4312** [comment]
Prior to July of this year we used a different debugger for Node.js programs. Now we switched to this one, which uses a "bootloader" given in the NODE_OPTIONS to attach to programs. This lets us do some cool things, like automatically debug child processes, build a better auto attach, have the debug terminal, etc. But it looks like that strategy causes a problem here.

**Comment by clarkttfu** [comment]
Thanks for the sharing.

**Comment by Busyrev** [comment]
Since VsCode 1v.60 Legacy node debuggers are gone: https://code.visualstudio.com/updates/v1_60#_javascript-debugging. And problem are still not solved for latest Node (17.3.0) and latest VsCode (1.63.2). The only available workaround is using [Deprecated] Node Debug (legacy): https://marketplace.visualstudio.com/items?itemName=ms-vscode.node-debug with ```"type": "legacy-node"``` in debug configuration. This works for me.

**Comment by connor4312** [comment]
@Busyrev you can also see my comment in https://github.com/microsoft/vscode-js-debug/issues/852#issuecomment-789900664 for a workaround
